### DISCUSSION TITLE
Composite HID with trackball/mouse support

### DISCUSSION THREAD

**Discussion by beastieboy** [OpenStickCommunity/GP2040-CE#437] [Ideas]
That would be great to have an option to enable additional HID device and connect old trackball or rotary controller visible as mouse.

**Comment by InfraredAces** [comment]
A reminder that this is first and foremost gamepad firmware.

**Comment by Bertzerker** [comment]
I agree, that better analog support would be great. Even if we need some additional parts like a adc or a second picopi. I'd like to make a smashbox style controller but with dual analog sticks.

**Reply by InfraredAces** [reply]
Perhaps you meant to post this on some other project. Right now, you can have dual analog sticks. Have a look at the documentation and look for the Analog Add-On.

**Reply by Bertzerker** [reply]
As far as I understood the addon menu options, analog will overwrite the dpad. But what do I know, I'll have a read through the docs. Thanks

**Reply by Bertzerker** [reply]
Thank you again. I just had a look at the docs and my autistic brain just can't read properly the first time. GP26 to 29 for the analog sticks and those override the analog emulation of the dpad input.

**Comment by embankment** [comment]
A mode for USB hosted trackball to analog joystick/joysticks _would_ be useful for Marble Madness or Katamari Damacy, among other games.

**Reply by Pelsin** [reply]
Sorry to necro-post, If the trackball registers as a generic mouse then this will be possible in the next release:
https://github.com/OpenStickCommunity/GP2040-CE/pull/1462

**Comment by TheTrainGoes** [comment]
Hi everyone,

This is general notice that we are closing down the discussion section of the GP2040-CE repo.  

If you would like to discuss anything further please open an issue or join our Discord:
https://discord.com/servers/openstickcommunity-1049366310389289001
